diff --git a/perlmod-test/src/pkg142.rs b/perlmod-test/src/pkg142.rs
index 167e54f..66a90a5 100644
--- a/perlmod-test/src/pkg142.rs
+++ b/perlmod-test/src/pkg142.rs
@@ -114,10 +114,6 @@ mod export {
         }
     }
 
-    #[export]
-    fn use_safe_putenv(on: bool) {
-        perlmod::ffi::use_safe_putenv(on);
-    }
 
     #[export]
     fn set_env(name: &str, value: &str) {
diff --git a/perlmod/src/ffi.rs b/perlmod/src/ffi.rs
index b1ac2e7..25b0352 100644
--- a/perlmod/src/ffi.rs
+++ b/perlmod/src/ffi.rs
@@ -412,8 +412,6 @@ extern "C" {
     pub fn RSPL_substr(orig: *mut SV, off: usize, len: usize) -> *mut SV;
 
     pub fn RSPL_defstash() -> *mut HV;
-
-    pub fn RSPL_set_use_safe_putenv(on: libc::c_int);
 }
 
 /// Argument marker for the stack.
@@ -580,8 +578,3 @@ where
     res
 }
 
-/// Tell perl to use a "safe" `putenv` call instead of manually manipulating the `environ`
-/// variable. Without this, changing environment variables can lead to crashes.
-pub fn use_safe_putenv(on: bool) {
-    unsafe { RSPL_set_use_safe_putenv(on as _) }
-}
diff --git a/perlmod/src/glue.c b/perlmod/src/glue.c
index 6e05684..acf3a53 100644
--- a/perlmod/src/glue.c
+++ b/perlmod/src/glue.c
@@ -454,9 +454,7 @@ extern HV* RSPL_defstash() {
     return PL_defstash;
 }
 
-extern void RSPL_set_use_safe_putenv(int on) {
-    PL_use_safe_putenv = on ? TRUE : FALSE;
-}
+
 
 /*
 These make are convoluted brainfarts:
