diff --git a/www/manager6/node/StatusView.js b/www/manager6/node/StatusView.js
index e05ec4df..c52f1d20 100644
--- a/www/manager6/node/StatusView.js
+++ b/www/manager6/node/StatusView.js
@@ -156,13 +156,6 @@ Ext.define('PVE.node.StatusView', {
             nodetab: 'aptrepositories',
         });
 
-        me.items.push({
-            xtype: 'pmxNodeInfoRepoStatus',
-            itemId: 'repositoryStatus',
-            product: 'Proxmox VE',
-            repoLink: `#${repoLink}`,
-        });
-
         me.callParent();
     },
 });
diff --git a/www/manager6/node/Summary.js b/www/manager6/node/Summary.js
index 9a3f84a4..a1e7424e 100644
--- a/www/manager6/node/Summary.js
+++ b/www/manager6/node/Summary.js
@@ -245,8 +245,6 @@ Ext.define('PVE.node.Summary', {
             },
         });
 
-        me.updateRepositoryStatus();
-
         me.callParent();
 
         let sp = Ext.state.Manager.getProvider();
