diff --git a/PVE/Ceph/Services.pm b/PVE/Ceph/Services.pm
index 4c3f3d99..1087f261 100644
--- a/PVE/Ceph/Services.pm
+++ b/PVE/Ceph/Services.pm
@@ -23,14 +23,14 @@ sub get_local_services {

         my $path = "/etc/systemd/system/ceph-$type.target.wants";
         my $regex = "ceph-$type\@(.*)\.service";
-        PVE::Tools::dir_glob_foreach(
-            $path,
-            $regex,
-            sub {
-                my (undef, $id) = @_;
-                $res->{$type}->{$id}->{service} = 1;
-            },
-        );
+
+        PVE::Tools::dir_glob_foreach($path, $regex, sub {
+            my (undef, $id) = @_;
+            my $keyring = "/var/lib/ceph/mon/ceph-$id/keyring";
+            return if !-e $keyring;    # treat as non-existent until mkfs has run
+
+            $res->{mon}->{$id} = { service => 1 };
+        });

         $path = "/var/lib/ceph/$type";
         $regex = "([^-]+)-(.*)";
