diff --git a/PVE/Tools.pm b/PVE/Tools.pm
index 948ef5e..6a18754 100644
--- a/PVE/Tools.pm
+++ b/PVE/Tools.pm
@@ -920,7 +920,7 @@ sub wait_for_vnc_port {
     my $starttime = [gettimeofday];
     my $elapsed;
 
-    my $cmd = ['/bin/ss', '-Htln', "sport = :$port"];
+    my $cmd = ['@sspath@/bin/ss', '-Htln', "sport = :$port"];
     push @$cmd, $family == AF_INET6 ? '-6' : '-4' if defined($family);
 
     my $found;
