diff --git a/PVE/Syscall.pm b/PVE/Syscall.pm
index 68e16fe..503d3fe 100644
--- a/PVE/Syscall.pm
+++ b/PVE/Syscall.pm
@@ -17,7 +17,7 @@ BEGIN {
         openat => &SYS_openat,
         close => &SYS_close,
         mkdirat => &SYS_mkdirat,
-        mknod => &SYS_mknod,
+        mknodat => &SYS_mknodat,
         faccessat => &SYS_faccessat,
         setresuid => &SYS_setresuid,
         fallocate => &SYS_fallocate,
diff --git a/PVE/Tools.pm b/PVE/Tools.pm
index 948ef5e..2077d68 100644
--- a/PVE/Tools.pm
+++ b/PVE/Tools.pm
@@ -1817,7 +1817,7 @@ sub mkdirat($$$) {
 
 sub mknod($$$) {
     my ($filename, $mode, $dev) = @_;
-    return syscall(PVE::Syscall::mknod, $filename, int($mode), int($dev)) == 0;
+    return syscall(PVE::Syscall::mknodat, $filename, int($mode), int($dev)) == 0;
 }
 
 sub fchownat($$$$$) {
