From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Fabian Ebner <f.ebner@proxmox.com>
Date: Fri, 4 Jun 2021 08:27:23 +0200
Subject: [PATCH] avoid passing deprecated 'upgrade' parameter

Signed-off-by: Fabian Ebner <f.ebner@proxmox.com>
---
 app/pve.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/pve.js b/app/pve.js
index 446b85d..dfff1b0 100644
--- a/app/pve.js
+++ b/app/pve.js
@@ -45,7 +45,7 @@ export default function PVEUI(UI){
 	    break;
 	case 'upgrade':
 	    url =  baseUrl + '/vncshell';
-	    params.upgrade = 1;
+	    params.cmd = 'upgrade';
 	    title = 'System upgrade on node ' + this.nodename;
 	    break;
 	case 'cmd':
