diff --git a/proxmox-ldap/Cargo.toml b/proxmox-ldap/Cargo.toml
index 902851f6..fe3315ea 100644
--- a/proxmox-ldap/Cargo.toml
+++ b/proxmox-ldap/Cargo.toml
@@ -14,5 +14,5 @@ ldap3 = { workspace = true, default-features = false, features = ["tls"] }
 serde = { workspace = true, features = ["derive"] }
 native-tls.workspace = true
 
-[dev_dependencies]
+[dev-dependencies]
 proxmox-async.workspace = true
